src/utils/include/utils/math/coordinate_transformation.hpp

```diff
@@ -19,57 +19,125 @@
 #ifndef UTILS_COORDINATE_TRANSFORMATION_HPP
 #define UTILS_COORDINATE_TRANSFORMATION_HPP
 
+/**
+ * @file
+ * Convert coordinates from the Cartesian system to the cylindrical system.
+ * The transformation functions are provided with three overloads:
+ * - one function for the trivial Cartesian <-> cylindrical transformation
+ * - one function to transform from/to a cylindrical system with custom axis
+ *   (extra @p axis argument, keep in mind the angle phi is under-defined)
+ * - one function to transform from/to an oriented cylindrical system with
+ *   custom axis (extra @p phi0 argument, the angle phi is well-defined)
+ */
+
 #include "utils/Vector.hpp"
 #include "utils/constants.hpp"
+#include "utils/math/interval.hpp"
 #include "utils/math/vec_rotate.hpp"
 
+#include <cmath>
+#include <tuple>
+
 namespace Utils {
 
-/** \brief Transform the given 3D position to cylinder coordinates with
- * longitudinal axis aligned with axis parameter.
+/**
+ * @brief Coordinate transformation from cylindrical to Cartesian coordinates.
+ * @param pos    %Vector to transform
  */
 inline Vector3d
-transform_coordinate_cartesian_to_cylinder(const Vector3d &pos,
-                                           const Vector3d &axis) {
+transform_coordinate_cartesian_to_cylinder(Vector3d const &pos) {
+  auto const r = std::sqrt(pos[0] * pos[0] + pos[1] * pos[1]);
+  auto const phi = std::atan2(pos[1], pos[0]);
+  return {r, phi, pos[2]};
+}
+
+/**
+ * @brief Coordinate transformation from cylindrical to Cartesian coordinates
+ * with change of basis.
+ * @param pos    %Vector to transform
+ * @param axis   Longitudinal axis of the cylindrical coordinates
+ * @param phi0   Reference angle for @f$ \phi = 0 @f$
```

Comment:
I'm not sure if this is clear looking at the function. You should explain what you mean by "Reference angle"